h/static/scripts/tests/controllers/character-limit-controller-test.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+var CharacterLimitController = require('../../controllers/character-limit-controller');
+var util = require('./util');
+
+describe('CharacterLimitController', function () {
+
+  /**
+   * Make a <textarea> with the character limit controller enhancement
+   * applied and return the various parts of the component.
+   */
+  function component(value, maxlength) {
+    maxlength = maxlength || 250;
+
+    var template = '<textarea class="js-character-limit" data-maxlength="';
+    template += maxlength + '">';
+    if (value) {
+      template += value;
+    }
+    template += '</textarea><span class="js-character-counter">Foo</span>';
+
+    var containerEl = util.setupComponent(document, template, {
+      '.js-character-limit': CharacterLimitController,
+    });
+    var counterEl = containerEl.querySelector('.js-character-counter');
+    var textarea = containerEl.querySelector('.js-character-limit');
+    var ctrl = textarea.controllers[0];
+
+    return {
+      containerEl: containerEl,
+      counterEl: counterEl,
+      textarea: textarea,
+      ctrl: ctrl,
+    };
+  }
```

Comment:
I'm using a helper function here instead of a `beforeEach()` because I have different tests that want to pass different args in to change _how the component gets created_. Creating test fixtures in `beforeEach()` just tightly couples tests to each other, I'd argue this is a better pattern.